src/config.rs

```diff
@@ -0,0 +1,75 @@
+//! # Config
+//! Crate to host the Configuration object for the bujo application
+
+use home::home_dir;
+use serde_derive::{Deserialize, Serialize};
+use std::fs;
+use std::path::PathBuf;
+
+#[derive(Serialize, Deserialize)]
+pub struct Config {
+    pub home: PathBuf,
+    pub bujorc: PathBuf,
+    pub data_dir: PathBuf,
+}
+
+/// The purpose of Config is to handle all references to config related items
+/// such as those relating to the .bujorc as well as to the data directories.
+impl Config {
+    /// Config::new() is intended to instantiate the Configuration of the current
+    /// application call by discovering the home directory for the operating system
+    /// and adding a reference to .bujorc.
+    pub fn new() -> Config {
+        let home_temp: PathBuf = home_dir().expect("Could not find home directory!");
+        let bujo_temp: PathBuf = [&home_temp.to_str().unwrap(), ".bujorc"].iter().collect();
```

Comment:
What does collect do here precisely? I understand it is used on iterables to effectively collect them into a new iterable. Is a PathBuf considered an iterable?

Reply:
My understanding is that collect() gathers the iterable into a collection (heap allocated data structure). In the case of PathBuf, it is an OsString which is a collection.

I didn't come up with that code though, it came from the docs [here](https://doc.rust-lang.org/std/path/struct.PathBuf.html#examples)

Reply:
```
OsString and OsStr bridge this gap by simultaneously representing Rust and platform-native string values, and in particular allowing a Rust string to be converted into an “OS” string with no cost if possible. A consequence of this is that OsString instances are not NUL terminated; in order to pass to e.g., Unix system call, you should create a CStr.
```

Interesting